### PR TITLE
job store embiggenings [BA-6166]

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+         to be widened. -->
+    <changeSet author="mcovarr" id="drop_job_store_simpleton_entry_fk" dbms="mysql,hsqldb,postgresql,mariadb">
+        <dropForeignKeyConstraint
+                baseTableName="JOB_STORE_SIMPLETON_ENTRY"
+                constraintName="FK_JOB_STORE_SIMPLETON_ENTRY_JOB_STORE_ENTRY_ID"
+        />
+    </changeSet>
+
+    <!-- BEGIN JOB_STORE_ENTRY PK widening -->
+    <changeSet author="mcovarr" id="enlarge_job_store_entry_id" dbms="mysql,hsqldb,mariadb">
```
Comment (grsterin) [inline database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml]: Looks like for Posgresql there're 2 changesets: one for modifying table column type, and another one for altering the autoincrementing sequence, whereas for other DBs there's a single `addAUtoIncrement` changeset. Does this single changeset do all the work for those databases (updating a column and sequence at once)?

Comment (mcovarr) [reply database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml]: That's right, the other databases can do the equivalent with a single changeset, as I previously learned [the hard way](https://github.com/broadinstitute/cromwell/blob/develop/database/migration/src/main/resources/changesets/enlarge_call_caching_hash_entry_id.xml#L6).

File: database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+                SELECT count(*)
+                FROM information_schema.sequences
+                WHERE sequence_name = 'JOB_STORE_ENTRY_JOB_STORE_ENTRY_ID_seq'
+                AND data_type = 'bigint';
+            </sqlCheck>
+        </preConditions>
+        <sql>alter sequence "JOB_STORE_ENTRY_JOB_STORE_ENTRY_ID_seq" as bigint;</sql>
+    </changeSet>
```
Comment (cjllanwarne) [inline database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml]: Could you comment why this precondition check is needed?

File: database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    <!-- Widen FK to match PK. -->
+    <changeSet author="mcovarr" id="enlarge_job_store_simpleton_entry_fk" dbms="mysql,hsqldb,postgresql,mariadb">
+        <modifyDataType
+                tableName="JOB_STORE_SIMPLETON_ENTRY"
+                columnName="JOB_STORE_ENTRY_ID"
+                newDataType="BIGINT"
+        />
+    </changeSet>
+
+    <!-- MariaDB's FK not null constraint does not survive the widening above and must be recreated explicitly. -->
```
Comment (cjllanwarne) [inline database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml]:
```suggestion
    <!-- MariaDB's FK NotNull constraint does not survive the widening above and must be recreated explicitly. -->
```

File: database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml (hunk 2 above)
Comment (mcovarr) [reply database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml]: I "borrowed" this from Khalid's earlier pg work but this appears to be an idempotency check (noop if the sequence is present and already consistent what the `alter` would do).

**Comment by cjllanwarne** [comment]
https://youtu.be/FcxsgZxqnEg?t=66

**Pull request merged**